lib/Adapter/DTL/DTLArgumentResolver.php

```diff
@@ -20,8 +20,8 @@ public function __construct(UpstreamArgumentResolver $dtlArgumnetResolver = null
         ], UpstreamArgumentResolver::ALLOW_UNKNOWN_ARGUMENTS | UpstreamArgumentResolver::MATCH_TYPE);
     }
 
-    public function resolveArguments(object $object, string $method, array $arguments): array
+    public function resolveArguments(object $object, string $method, array $arguments, array $extraArgs): array
     {
-        return $this->dtlArgumnetResolver->resolveArguments(get_class($object), $method, $arguments);
+        return $this->dtlArgumnetResolver->resolveArguments(get_class($object), $method, array_merge($arguments, $extraArgs));
```

Comment:
Separated the "real" request arguments from the "extra" ones. Ideally we'd refactor - the only "extra" should be the AMP cancellation token.